dandiapi/api/tasks/scheduled.py

```diff
@@ -29,6 +32,19 @@
 logger = get_task_logger(__name__)
 
 
+def throttled_iterator(iterable: Iterable, chunk_size: int = 100, sleep_time: int = 1) -> Iterable:
+    """
+    Yield items from an iterable, with a sleep between chunks.
+
+    This is useful for putting messages on a queue, where you want to avoid
+    overwhelming the queue with too many messages at once.
+    """
+    for chunk in ichunked(iterable, chunk_size):
+        for item in chunk:
+            yield item
+            time.sleep(sleep_time / chunk_size)
```

Comment:
I'm confused by this. As chunk size increases, sleep time decreases? Shouldn't it be the other way around, where as your chunks get bigger, you sleep for longer?

Reply:
Oops, nevermind, I misread this initially and thought that the sleep was happening between chunks. I can see now it's ensuring that each chunk takes at least `sleep_time` number of seconds to complete.